### DISCUSSION TITLE
Improve Random Dinner Selection Feature in Meal Planner

### DISCUSSION THREAD

**Discussion by dandud100** [mealie-recipes/mealie#3679] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

When I use the Meal Planner to plan a random dinner, it sometimes selects the exact same dish I had yesterday. If I press the button multiple times, it can insert the same dish twice or more in a row. Additionally, I find it inconvenient that it's possible to plan more than one dinner for a day using the random button.

![image](https://github.com/mealie-recipes/mealie/assets/43549788/8618b10b-0f71-4c88-820c-bcbdd6be2c28)

### Please provide a concise description of the feature that would resolve your issue.

It would be better if the button provided a suggestion in a separate box, allowing the user to either regenerate or accept the suggestion. As it stands now, I have to press the button, delete the recipe, and then press the button again, which is not user-friendly.

### Please consider and list out some caveats or tradeoffs made in your design decision

-

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by paulhugill** [comment]
For my use case, I would be happy if this was controllable via a meal planner rule.
There is the `Last Made` option for a rule, but this seems to require a specific date and I am not going to change that every week and not even sure if I would use the `I made this` feature.

Instead, if there was a `Last in Plan` and allowing someone to select the number of days since it was in a plan, then people could choose whether they want that to be 0 days (can have it the next day), 7 days to not have it in the same week or whatever fits for them.

**Reply by michael-genson** [reply]
Agreed that requiring a static date is a major limitation for the last made filter. There was some discussion around improving this, can't remember if there was ever a PR or not.

FYI whenever a recipe is added to the meal planner, on that date it gets marked as "made" so you wouldn't have to worry about a separate "last in plan" date.

**Reply by paulhugill** [reply]
Great that it gets marked as made, thanks for pointing that out, I'm new to Mealie and just figuring out how to use it and make it work for me. I've got it hooked into Home Assistant and have an automation generating the next 7 days of random meals on the plan, but it seems to pick the same 2 meals for the whole week!

I did spot this FR after commenting too, which is basically that: https://github.com/mealie-recipes/mealie/discussions/4717
